tests/e2e/support/objects/app-files/page/public.ts

```diff
@@ -15,20 +20,14 @@ export class Public {
   }
 
   async authenticate({ password }: { password: string }): Promise<void> {
-    await this.#page.locator('input[type="password"]').fill(password)
-    await this.#page
-      .locator(
-        '//*[@id="password-submit"]|//*[@id="oc-textinput-3"]/ancestor::div[contains(@class, "oc-mb-s")]/following-sibling::button'
-      )
-      .click()
+    await this.#page.locator(passwordInput).fill(password)
+    await this.#page.locator(publicLinkAuthorizeButton).click()
   }
 
   async upload({ resources }: { resources: File[] }): Promise<void> {
     const startUrl = this.#page.url()
-    const resourceSelector = `[data-test-resource-name="%s"]`
-    await this.#page
-      .locator('//input[@id="files-file-upload-input"]')
-      .setInputFiles(resources.map((file) => file.path))
+    const resourceSelector = resourceNameSelector
+    await this.#page.locator(fileUploadInput).setInputFiles(resources.map((file) => file.path))
     const names = resources.map((file) => path.basename(file.name))
     await Promise.all(
       names.map((name) => this.#page.waitForSelector(util.format(resourceSelector, name)))
```

Comment:
might not need the extra variable
```suggestion
    await this.#page.locator(fileUploadInput).setInputFiles(resources.map((file) => file.path))
    const names = resources.map((file) => path.basename(file.name))
    await Promise.all(
      names.map((name) => this.#page.waitForSelector(util.format(resourceNameSelector, name)))
```